pkg/syncer/spec/spec_controller.go

```diff
@@ -223,3 +235,44 @@ func (c *Controller) processNextWorkItem(ctx context.Context) bool {
 
 	return true
 }
+
+func newSecretLister(secretIndexer cache.Indexer) specmutators.ListSecretFunc {
+	return func(clusterName logicalcluster.Name, namespace string) ([]*corev1.Secret, error) {
+		secretList, err := secretIndexer.ByIndex(byWorkspaceAndNamespaceIndexName, workspaceAndNamespaceIndexKey(clusterName, namespace))
+		if err != nil {
+			return nil, fmt.Errorf("error listing secrets for workspace %s: %w", clusterName, err)
+		}
+
+		secrets := make([]*corev1.Secret, 0, len(secretList))
+		for _, elem := range secretList {
+			var secret corev1.Secret
+
+			unstrSecret := elem.(*unstructured.Unstructured)
+			if err = runtime.DefaultUnstructuredConverter.FromUnstructured(unstrSecret.Object, &secret); err != nil {
```

Comment:
feeling uncomfortable decoding a (potentially long) list of secrets, just to find one with an annotation.

Reply:
I see nothing in the consuming code that cannot be done on unstructured.